### PR TITLE
feat: solana standard tx parsing

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

Solana unchained-client support (parser and api client)

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/7738

## Risk

> High Risk PRs Require 2 approvals

Low - behind feature flag and only basic send/receive sol support (no spl tokens)

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing

- Unit tests should work

### Engineering

:point_up:

### Operations

- [x] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

N/A

### CODE REVIEW

File: packages/unchained-client/src/solana/parser/index.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        if (result) return result
+      }
+    })()
+
+    const parsedTx: ParsedTx = {
+      address,
+      blockHeight: tx.blockHeight,
+      blockTime: tx.timestamp,
+      chainId: this.chainId,
+      confirmations: 1,
```
Comment (0xApotheosis) [inline packages/unchained-client/src/solana/parser/index.ts:51]: Should this be `confirmations: tx.confirmations,`?

```suggestion
      confirmations: tx.confirmations,
```

**Review by 0xApotheosis — COMMENTED** [review]
Looks good, just one query about the confirmations.

File: packages/unchained-client/src/solana/parser/index.ts (hunk 1 above)
Comment (NeOMakinG) [reply packages/unchained-client/src/solana/parser/index.ts:51]:
```
[
  {
    "description": "Human readable interpretation of the transaction",
    "type": "UNKNOWN",
    "source": "FORM_FUNCTION",
    "fee": 5000,
    "feePayer": "8cRrU1NzNpjL3k2BwjW3VixAcX6VFc29KHr4KZg8cs2Y",
    "signature": "yy5BT9benHhx8fGCvhcAfTtLEHAtRJ3hRTzVL16bdrTCWm63t2vapfrZQZLJC3RcuagekaXjSs2zUGQvbcto8DK",
    "slot": 148277128,
    "timestamp": 1656442333,
    "nativeTransfers": [
      {
        "fromUserAccount": "text",
        "toUserAccount": "text"
      }
    ],
    "tokenTransfers": [
      {
        "fromUserAccount": "text",
        "toUserAccount": "text",
        "fromTokenAccount": "text",
        "toTokenAccount": "text",
        "tokenAmount": 0,
        "mint": "DsfCsbbPH77p6yeLS1i4ag9UA5gP9xWSvdCx72FJjLsx"
      }
    ],
    "accountData": [
      {
        "account": "text",
        "nativeBalanceChange": 0,
        "tokenBalanceChanges": [
          {
            "userAccount": "F54ZGuxyb2gA7vRjzWKLWEMQqCfJxDY1whtqtjdq4CJ",
            "tokenAccount": "2kvmbRybhrcptDnwyNv6oiFGFEnRVv7MvVyqsxkirgdn",
            "mint": "DUSTawucrTsGU8hcqRdHDCbuYhCPADMLM2VcCb8VnFnQ",
            "rawTokenAmount": {
              "tokenAmount": "text"
            }
          }
        ]
      }
    ],
    "transactionError": {
      "error": "text"
    },
    "instructions": [
      {
        "accounts": [
          "8uX6yiUuH4UjUb1gMGJAdkXorSuKshqsFGDCFShcK88B"
        ],
        "data": "kdL8HQJrbbvQRGXmoadaja1Qvs",
        "programId": "MEisE1HzehtrDpAAT8PnLHjpSSkRYakotTuJRPjTpo8",
        "innerInstructions": [
          {
            "accounts": [
              "text"
            ],
            "data": "text",
            "programId": "text"
          }
        ]
      }
    ],
    "events": {
      "nft": {
        "description": "text",
        "type": "NFT_SALE",
        "source": "FORM_FUNCTION",
        "amount": 1000000,
        "fee": 5000,
        "feePayer": "8cRrU1NzNpjL3k2BwjW3VixAcX6VFc29KHr4KZg8cs2Y",
        "signature": "4jzQxVTaJ4Fe4Fct9y1aaT9hmVyEjpCqE2bL8JMnuLZbzHZwaL4kZZvNEZ6bEj6fGmiAdCPjmNQHCf8v994PAgDf",
        "slot": 148277128,
        "timestamp": 1656442333,
        "saleType": "AUCTION",
        "buyer": "text",
        "seller": "text",
        "staker": "text",
        "nfts": [
          {
            "mint": "DsfCsbbPH77p6yeLS1i4ag9UA5gP9xWSvdCx72FJjLsx",
            "tokenStandard": "NonFungible"
          }
        ]
      },
      "swap": {
        "nativeInput": {
          "account": "2uySTNgvGT2kwqpfgLiSgeBLR3wQyye1i1A2iQWoPiFr",
          "amount": "100000000"
        },
        "nativeOutput": {
          "account": "2uySTNgvGT2kwqpfgLiSgeBLR3wQyye1i1A2iQWoPiFr",
          "amount": "100000000"
        },
        "tokenInputs": [
          {
            "userAccount": "F54ZGuxyb2gA7vRjzWKLWEMQqCfJxDY1whtqtjdq4CJ",
            "tokenAccount": "2kvmbRybhrcptDnwyNv6oiFGFEnRVv7MvVyqsxkirgdn",
            "mint": "DUSTawucrTsGU8hcqRdHDCbuYhCPADMLM2VcCb8VnFnQ",
            "rawTokenAmount": {
              "tokenAmount": "text"
            }
          }
        ],
        "tokenOutputs": [
          {
            "userAccount": "F54ZGuxyb2gA7vRjzWKLWEMQqCfJxDY1whtqtjdq4CJ",
            "tokenAccount": "2kvmbRybhrcptDnwyNv6oiFGFEnRVv7MvVyqsxkirgdn",
            "mint": "DUSTawucrTsGU8hcqRdHDCbuYhCPADMLM2VcCb8VnFnQ",
            "rawTokenAmount": {
              "tokenAmount": "text"
            }
          }
        ],
        "tokenFees": [
          {
            "userAccount": "F54ZGuxyb2gA7vRjzWKLWEMQqCfJxDY1whtqtjdq4CJ",
            "tokenAccount": "2kvmbRybhrcptDnwyNv6oiFGFEnRVv7MvVyqsxkirgdn",
            "mint": "DUSTawucrTsGU8hcqRdHDCbuYhCPADMLM2VcCb8VnFnQ",
            "rawTokenAmount": {
              "tokenAmount": "text"
            }
          }
        ],
        "nativeFees": [
          {
            "account": "2uySTNgvGT2kwqpfgLiSgeBLR3wQyye1i1A2iQWoPiFr",
            "amount": "100000000"
          }
        ],
        "innerSwaps": [
          {
            "tokenInputs": [
              {
                "fromUserAccount": "text",
                "toUserAccount": "text",
                "fromTokenAccount": "text",
                "toTokenAccount": "text",
                "tokenAmount": 0,
                "mint": "DsfCsbbPH77p6yeLS1i4ag9UA5gP9xWSvdCx72FJjLsx"
              }
            ],
            "tokenOutputs": [
              {
                "fromUserAccount": "text",
                "toUserAccount": "text",
                "fromTokenAccount": "text",
                "toTokenAccount": "text",
                "tokenAmount": 0,
                "mint": "DsfCsbbPH77p6yeLS1i4ag9UA5gP9xWSvdCx72FJjLsx"
              }
            ],
            "tokenFees": [
              {
                "fromUserAccount": "text",
                "toUserAccount": "text",
                "fromTokenAccount": "text",
                "toTokenAccount": "text",
                "tokenAmount": 0,
                "mint": "DsfCsbbPH77p6yeLS1i4ag9UA5gP9xWSvdCx72FJjLsx"
              }
            ],
            "nativeFees": [
              {
                "fromUserAccount": "text",
                "toUserAccount": "text"
              }
            ],
            "programInfo": {
              "source": "ORCA",
              "account": "whirLbMiicVdio4qvUfM5KAg6Ct8VwpYzGff3uctyCc",
              "programName": "ORCA_WHIRLPOOLS",
              "instructionName": "whirlpoolSwap"
            }
          }
        ]
      }
    }
  }
]
```

Either regular `getParsedTransaction` or Helius one returns doesn't have `confirmations`, I think we don't have this information but setting`confirmations` to 1 seems off 

maybe we should at least add a comment to explain this magic number choice?

**Comment by NeOMakinG** [comment]
@kaladinlight seems very sane to me, I could stamp it but when do we plan to parse the SPL Tokens transfers? We would need to create an issue or plan the follow up so we don't lose track

File: packages/unchained-client/src/solana/parser/index.ts (hunk 1 above)
Comment (kaladinlight) [reply packages/unchained-client/src/solana/parser/index.ts:51]: As neo explained, we don't have confirmations from the endpoint, but we are also only handling finalized transacitons, no concept of mempool as far as we are concerned. I chose 1 just as a positive number in the case of `>0` checks for status. I will add a comment.

**Comment by kaladinlight** [comment]
> @kaladinlight seems very sane to me, I could stamp it but when do we plan to parse the SPL Tokens transfers? We would need to create an issue or plan the follow up so we don't lose track

Was starting with standard send full implementation first and then following up with tokens. Will fall into this guy: https://github.com/shapeshift/web/issues/7903. I will update to be more generic

File: packages/unchained-client/src/solana/parser/index.ts (hunk 1 above)
Comment (kaladinlight) [reply packages/unchained-client/src/solana/parser/index.ts:52]: 4ab1c60423c8a42b1a79db32cca9269a670e038d

Comment (0xApotheosis) [reply packages/unchained-client/src/solana/parser/index.ts:52]: If the key/concept of `confirmations` doesn't exist in this domain would it be better to have `undefined` instead, or is that too pedantic? Albeit accepting the additional plumbing of `confirmations: number | undefined`.

**Review by 0xApotheosis — APPROVED** [review]
The conversation around `confirmations` is only a nitpick, and not worth holding this PR up for - let's get 'er in!

**Pull request merged**